src/vmm/src/devices/virtio/virtio_block/device.rs

```diff
@@ -95,17 +97,44 @@ impl DiskProperties {
             );
         }
 
+        Ok(disk_size)
+    }
+
+    /// Create a new file for the block device using a FileEngine
+    pub fn new(
+        disk_image_path: String,
+        is_disk_read_only: bool,
+        file_engine_type: FileEngineType,
+    ) -> Result<Self, VirtioBlockError> {
+        let mut disk_image = Self::open_file(&disk_image_path, is_disk_read_only)?;
+        let disk_size = Self::file_size(&disk_image_path, &mut disk_image)?;
         let image_id = Self::build_disk_image_id(&disk_image);
 
         Ok(Self {
             file_path: disk_image_path,
             file_engine: FileEngine::from_file(disk_image, file_engine_type)
                 .map_err(VirtioBlockError::FileEngine)?,
+            is_disk_read_only,
             nsectors: disk_size >> SECTOR_SHIFT,
             image_id,
         })
     }
 
+    /// Update the path to the file backing the block device
+    pub fn update(&mut self, disk_image_path: String) -> Result<(), VirtioBlockError> {
+        let mut disk_image = Self::open_file(&disk_image_path, self.is_disk_read_only)?;
+        let disk_size = Self::file_size(&disk_image_path, &mut disk_image)?;
+
+        self.image_id = Self::build_disk_image_id(&disk_image);
+        self.file_engine
+            .update_file_path(disk_image)
```

Comment:
What if we move `file_engine` out of `DiskProperties` into block struct itself? I feel `DiskProperties` is not really a place for it.

Reply:
I thought about it. But I think, this is refactoring :) I'd like to just fix the problem without restructuring the logic of the implementation.

Reply:
Ok, can you create an issue to refactor this so we will not forget?